sbin/build.sh

```diff
@@ -535,6 +535,9 @@ createSourceArchive() {
     # Transform 'OpenJDK11U-jdk_aarch64_linux_hotspot_11.0.12_7.tar.gz' to 'OpenJDK11U-sources_11.0.12_7.tar.gz'
     # shellcheck disable=SC2001
     srcArchiveName="$(echo "${BUILD_CONFIG[TARGET_FILE_NAME]}" | sed 's/-jdk_.*_temurin_/-sources_/g')"
+  elif
```

Comment:
For JDK17, the file names being provided to this function does not contain the "temurin" string in order to hit the first clause, as the sources for release are built alongside the X64 linux hotspot build, the file names have the following format OpenJDK17U-sources_x64_linux_hotspot_2022-05-15-12-05.tar.gz. The second clause is to allow for this whereby the temurin name isnt included/constructed as part of the target file name parameter.

Reply:
The same naming convention is used across the JDK11 & JDK18 build pipelines also as seen in the most recent source artifacts here https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk18-pipeline/lastSuccessfulBuild/artifact/target/linux/x64/temurin/ & here https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk11-pipeline/lastSuccessfulBuild/artifact/target/linux/x64/temurin/

Reply:
I'm not sure we ever produce filenames with `temurin` in the name, so I suspect we should just adjust the above `if` statement to use `_temurin` and not have this `else` case (or just use wildcards e.g. -jdk.*_[a-zA-Z0-9]*_ for the search string

Reply:
FYI This was introduced by me in the hotspot->temurin switch a few months back: https://github.com/adoptium/temurin-build/pull/2818/files so I'm good with changing this line instead of the elif